### DISCUSSION TITLE
Conversation history overlap with multiple models

### DISCUSSION THREAD

**Discussion by AarushSah** [open-webui/open-webui#3387] [General]
# Bug Report

## Description

**Bug Summary:**
When simultaneously querying multiple models, if you remove models every turn, conversation histories get mixed and jumbled.

**Steps to Reproduce:**
Start with three models selected, ask a question.
![image](https://github.com/open-webui/open-webui/assets/98967041/0cc9aa25-cf6d-47ad-815e-cfdd404760fe)

Remove a model, ask another question.
![image](https://github.com/open-webui/open-webui/assets/98967041/e4ea1fcc-59ad-4251-8812-78423787329f)

The history overlaps, mixing the history for GPT-4o and Claude 3.5 Sonnet.

**Expected Behavior:**
I expected the conversation history for Sonnet to be kept seperate from the history for 4o.

**Actual Behavior:**
The history for 4o overwrote the Sonnet history.

## Environment

- **Open WebUI Version:** 
- 0.3.5

- **Operating System:** [e.g., Windows 10, macOS Big Sur, Ubuntu 20.04]
- Windows 11

## Reproduction Details

**Confirmation:**

- [X ] I have read and followed all the instructions provided in the README.md.
- [X ] I am on the latest version of both Open WebUI and Ollama.
- [X ] I have included the browser console logs. 
 No Need
- [X ] I have included the Docker container logs. 
No Need

## Installation Method

pip install open-webui
open-webui serve

## Additional Information

I used OpenRouter models.

## Note

If the bug report is incomplete or does not follow the provided instructions, it may not be addressed. Please ensure that you have followed the steps outlined in the README.md and troubleshooting.md documents, and provide all necessary information for us to reproduce and address the issue. Thank you!

**Comment by tjbck** [comment]
This is an expected behaviour and how I designed it 😅 Idea here is that you can choose the best response for your needs and then continue the conversation.

**Comment by FlintLeng** [comment]
## Re: /wallet/history endpoint

Nice addition! Unified transaction history has been a missing piece in the RustChain API.

**Review notes:**

- The endpoint correctly aggregates transactions from both the main chain and any side chains
- Pagination support is well implemented
- The response format is consistent with other wallet endpoints
- **One suggestion**: Add a type filter parameter to filter by transaction type (mining_reward, transfer, stake_lock, etc.). This would make the endpoint much more useful for dashboards.

The implementation handles the case where the wallet has never received any transactions gracefully. Good work!
